src/history/recorder.test.js

```diff
@@ -58,66 +56,157 @@ describe('Recorder', () => {
+    it('returns the file path for the given service provider’s document type', async () => {
+      expect(subject.getPathFor(SERVICE_PROVIDER_ID, TYPE)).to.equal(EXPECTED_FILE_PATH);
+    });
+  });
 
-    context('when this is the first record', () => {
-      it('returns a boolean to specify this is the first one', () => {
-        expect(isFirstRecord).to.equal(true);
+  describe('#isTracked', () => {
+    after(resetGitRepository);
+
+    context('when the file does not exists', () => {
+      it('returns false', async () => {
+        expect(await subject.isTracked(SERVICE_PROVIDER_ID, TYPE)).to.be.false;
       });
     });
 
-    context('when this is not the first record', () => {
-      it('returns a boolean to specify this is not the first one', async () => {
-        const recordResult = await subject.record({
+    context('when the file already exists', () => {
+      before(async () => {
+        await subject.record({
           serviceId: SERVICE_PROVIDER_ID,
           documentType: TYPE,
-          content: PERSIST_FILE_CONTENT,
-          isFiltered: false
+          content: FILE_CONTENT,
+          changelog: 'Start tracking document'
         });
-        expect(recordResult.isFirstRecord).to.equal(false);
       });
+
+      it('returns true', async () => {
+        expect(await subject.isTracked(SERVICE_PROVIDER_ID, TYPE)).to.be.true;
+      });
+    });
+  });
+
+  describe('#_getCommits', () => {
+    let firstRecordId;
+    let secondRecordId;
+    before(async () => {
+      let record = await subject.record({
+        serviceId: SERVICE_PROVIDER_ID,
+        documentType: TYPE,
+        content: FILE_CONTENT,
+      });
+      firstRecordId = record.id;
+      record = await subject.record({
+        serviceId: SERVICE_PROVIDER_ID,
+        documentType: TYPE,
+        content: `${FILE_CONTENT} (with additional content to trigger a record)`,
+      });
+      secondRecordId = record.id;
+    });
+
+    after(resetGitRepository);
+
+    it('returns all commits related to the given file path', async () => {
+      const filePath = subject.getPathFor(SERVICE_PROVIDER_ID, TYPE);
+      const commits = await subject._getCommits(filePath);
+      expect(commits).to.have.length(2);
+      expect(commits[0].hash).to.include(secondRecordId);
+      expect(commits[1].hash).to.include(firstRecordId);
+    });
+  });
+
+  describe('#getLatestRecord', () => {
+    let lastSnapshotId;
+    let lastestRecord;
+
+    before(async () => {
+      await subject.record({
+        serviceId: SERVICE_PROVIDER_ID,
+        documentType: TYPE,
+        content: FILE_CONTENT,
+      });
+      const { id: recordId } = await subject.record({
+        serviceId: SERVICE_PROVIDER_ID,
+        documentType: TYPE,
+        content: `${FILE_CONTENT} (with additional content to trigger a record)`,
+      });
+      lastSnapshotId = recordId;
+      lastestRecord = await subject.getLatestRecord(SERVICE_PROVIDER_ID, TYPE);
```

Comment:
Either `latest` (= _plus récent_) or `last` (= _dernier_), but `lastest` doesn't exist 😉 
```suggestion
      latestRecord = await subject.getLatestRecord(SERVICE_PROVIDER_ID, TYPE);
```